src/main/java/org/snomed/snowstorm/core/data/services/ConceptChangeHelper.java

```diff
@@ -0,0 +1,376 @@
+                        .must(termsQuery(ReferenceSetMember.Fields.REFERENCED_COMPONENT_ID, synonymAndTextDefIds))
+                        .must(termsQuery(ReferenceSetMember.LanguageFields.ACCEPTABILITY_ID_FIELD_PATH, Concepts.PREFERRED)))
+                .withFields(ReferenceSetMember.Fields.REFERENCED_COMPONENT_ID)
+                .build();
+        try (final SearchHitsIterator<ReferenceSetMember> stream = elasticsearchTemplate.searchForStream(languageMemberQuery, ReferenceSetMember.class)) {
+            stream.forEachRemaining(hit -> preferredDescriptionIds.add(parseLong(hit.getContent().getReferencedComponentId())));
+        }
+
+        Set<Long> changedComponents = referenceComponentIdToConceptMap.keySet()
+                .stream()
+                .filter(r -> preferredDescriptionIds.contains(r) || !synonymAndTextDefIds.contains(r))
+                .collect(Collectors.toSet());
+
+        for (Long componentId : changedComponents) {
+            changedConcepts.add(referenceComponentIdToConceptMap.get(componentId));
+        }
+
+        logger.info("Change report complete for branch {} time range {} to {}", path, start, end);
+
+        return changedConcepts;
+
+    }
+
+    private NativeSearchQueryBuilder componentsReplacedCriteria(Set<String> versionsReplaced, String... limitFieldsFetched) {
+        NativeSearchQueryBuilder builder = new NativeSearchQueryBuilder()
+                .withQuery(boolQuery()
+                        .must(termsQuery("_id", versionsReplaced))
+                        .mustNot(getNonStatedRelationshipClause()))
+                .withPageable(LARGE_PAGE);
+        if (limitFieldsFetched.length > 0) {
+            builder.withFields(limitFieldsFetched);
+        }
+        return builder;
+    }
+
+    private QueryBuilder getNonStatedRelationshipClause() {
+        return termsQuery(Relationship.Fields.CHARACTERISTIC_TYPE_ID, Concepts.INFERRED_RELATIONSHIP, Concepts.ADDITIONAL_RELATIONSHIP);
+    }
+
+    private NativeSearchQueryBuilder newSearchQuery(BoolQueryBuilder branchUpdatesCriteria) {
+        return new NativeSearchQueryBuilder()
+                .withQuery(boolQuery()
+                        .must(branchUpdatesCriteria)
+                        .mustNot(getNonStatedRelationshipClause()))
+                .withPageable(LARGE_PAGE);
+    }
+
+    /**
+     * Return Concept identifiers that have been deleted on one Branch, with any of the Concepts' Descriptions being modified on
+     * the other Branch.
+     *
+     * @param source The Branch to find either deleted Concepts or modified Descriptions.
+     * @param target The Branch to find either deleted Concepts or modified Descriptions.
+     * @return Concept identifiers that have been deleted on one Branch, with any of the Concepts' Descriptions being modified on
+     * the other Branch.
+     */
+    public Set<Long> getContradictoryChanges(Branch source, Branch target) {
+        Set<String> contradictoryChanges = new HashSet<>();
```

Comment:
It is hard to name this method correctly in our discussion. We have named this with "contradictoryChanges" based on there might be additional cases in the future. Currently this method effectively can be renamed as "  getConceptsDeletedByOneWhileWithDescriptionsModifiedByTheOther" which is a bit long. Maybe we could name this as getConceptsWithContradictoryChanges()